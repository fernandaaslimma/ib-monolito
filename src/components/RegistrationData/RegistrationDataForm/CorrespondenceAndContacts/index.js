import CorrespondenceAndContacts from "./CorrespondenceAndContacts";

export default CorrespondenceAndContacts;
