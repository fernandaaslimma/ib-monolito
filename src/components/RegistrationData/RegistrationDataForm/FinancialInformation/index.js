import FinancialInformation from "./FinancialInformation";

export default FinancialInformation;
