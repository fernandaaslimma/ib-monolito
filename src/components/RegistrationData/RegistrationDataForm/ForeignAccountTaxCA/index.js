import ForeignAccountTaxCA from "./ForeignAccountTaxCA";

export default ForeignAccountTaxCA;
