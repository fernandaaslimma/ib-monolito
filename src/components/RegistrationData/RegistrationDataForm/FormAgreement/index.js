import FormAgreement from "./FormAgreement";

export default FormAgreement;
