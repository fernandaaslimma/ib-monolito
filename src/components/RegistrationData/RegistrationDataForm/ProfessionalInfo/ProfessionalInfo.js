import React, { Fragment, useContext } from "react";
import { translate } from "../../../../utils/i18n";
import { validateInputByType } from "../../../../utils/validations/input";
import SectionedFormContent from "../../../common/SectionedFormContent";
import { Dropdown, Input, AlertMessage } from "react-bocombbm-components";
import {
  GridWrapper,
  InnerGridWrapper,
  Block,
  InnerBlock,
  Item
} from "../styles";
import { PROFESSIONAL_INFORMATION } from "../../../../utils/constants";
import { professionList } from "../../../../utils/constantLists";
import { StateContext } from "../RegistrationDataForm";
import resetObjectFields from "../../../../utils/resetObjectFields";

function ProfessionalInfo({ onChange, validateSection, disabled }) {
  const { temporaryInvalidFields, isFilled, formModel, closeAll } = useContext(
    StateContext
  );
  const invalidFields = temporaryInvalidFields.professionalInformation;
  let { professionalInformation } = formModel.content;

  const { activity } = professionalInformation;

  if (activity === "retired") {
    resetObjectFields(
      ["company", "otherActivitySpecified", "admissionDate"],
      professionalInformation
    );
  } else if (
    activity === "privateSectorEmployee" ||
    activity === "publicSectorEmployee"
  ) {
    resetObjectFields(
      ["otherActivitySpecified", "admissionDate"],
      professionalInformation
    );
  } else if (
    activity === "selfEmployed" ||
    activity === "others" ||
    activity === "professional" ||
    activity === "socioOwner"
  ) {
    resetObjectFields(
      ["company", "occupation", "admissionDate"],
      professionalInformation
    );
  } else if (activity === "fromHome") {
    resetObjectFields(
      ["company", "otherActivitySpecified", "occupation", "admissionDate"],
      professionalInformation
    );
  } else if (activity === "owner") {
    resetObjectFields(
      ["otherActivitySpecified", "occupation", "admissionDate"],
      professionalInformation
    );
  }

  const changeActivity = e => {
    const value = e.value;
    professionalInformation.activity = value;

    onChange(professionalInformation, PROFESSIONAL_INFORMATION);
  };

  const changeInput = (e, field, fieldDeep) => {
    const value = e.target.value;

    fieldDeep
      ? (professionalInformation[field][fieldDeep] = value)
      : (professionalInformation[field] = value);

    onChange(professionalInformation, PROFESSIONAL_INFORMATION);
  };

  const defineClosedDescription = activity => {
    switch (activity) {
      case "retired":
        return `${translate(`${activity}_DESCRIPTION`.toUpperCase())} ${
          professionalInformation.occupation
        }`;
      case "privateSectorEmployee":
        return `${translate(`${activity}_DESCRIPTION`.toUpperCase())} ${
          professionalInformation.occupation
        } ${translate("AT_COMPANY")} ${professionalInformation.company.name}`;
      case "publicSectorEmployee":
        return `${translate(`${activity}_DESCRIPTION`.toUpperCase())} ${
          professionalInformation.occupation
        } ${translate("AT_COMPANY")} ${professionalInformation.company.name}`;
      case "selfEmployed":
        return `${translate(`${activity}_DESCRIPTION`.toUpperCase())}`;
      case "fromHome":
        return `${translate(`${activity}_DESCRIPTION`.toUpperCase())}`;
      case "owner":
        return `${translate(`${activity}_DESCRIPTION`.toUpperCase())} ${
          professionalInformation.company.name
        }`;
      case "others":
      case "professional":
      case "socioOwner":
        return `${translate("OTHERS_DESCRIPTION")} ${
          professionalInformation.otherActivitySpecified !== null
            ? professionalInformation.otherActivitySpecified
            : ""
        }`;
      default:
        return translate("PROFESSIONAL_INFO");
    }
  };

  const renderInvalidMessage = () => {
    const renderGroup = collection =>
      collection.map(item => (
        <Item data-last={translate("ATUCAD_AND")} key={item}>
          {translate(item.toUpperCase())}
        </Item>
      ));
    const keys = Object.keys(invalidFields);
    return keys.map(item => renderGroup(invalidFields[item]));
  };

  return (
    <SectionedFormContent
      openedDescription={translate("PROFESSIONAL_INFO").toUpperCase()}
      dataTest="SectionedProfessionalInfo"
      closedDescription={defineClosedDescription(
        professionalInformation.activity
      )}
      callback={() =>
        validateSection(professionalInformation, PROFESSIONAL_INFORMATION)
      }
      valid={isFilled.professionalInformation}
      close={closeAll}
    >
      <GridWrapper>
        {invalidFields && (
          <Block span={[1, 3]}>
            <AlertMessage type="error" icon="Attention">
              {translate("REQUIRED_FIELDS_OMMITED")}:<br />
              {renderInvalidMessage()}
            </AlertMessage>
          </Block>
        )}
        <Block span={[1, 3]}>
          <InnerGridWrapper columns={2}>
            <InnerBlock IEFractionParts={2} layered={2}>
              <Dropdown
                classNamePrefix="DropdownActivity"
                onChange={changeActivity}
                name="Activities"
                options={professionList}
                value={professionList.find(
                  item => item.value === professionalInformation.activity
                )}
                label={translate("ACTIVITY")}
                dataTest="activitySelection"
                disabled={disabled}
              />
            </InnerBlock>
            {professionalInformation.activity === "retired" && (
              <InnerBlock IEFractionParts={2}>
                <Input
                  dataTest="InputOccupation"
                  onChange={e => changeInput(e, "occupation")}
                  value={professionalInformation.occupation}
                  label={translate("OCCUPATION")}
                  required
                  valid={validateInputByType["default"](
                    professionalInformation.occupation
                  )}
                  disabled={disabled}
                />
              </InnerBlock>
            )}

            {professionalInformation.activity === "owner" && (
              <InnerBlock IEFractionParts={2}>
                <Input
                  dataTest="InputCompanyName"
                  onChange={e => changeInput(e, "company", "name")}
                  value={professionalInformation.company.name}
                  label={translate("COMPANY")}
                  required
                  valid={validateInputByType["default"](
                    professionalInformation.company.name
                  )}
                  disabled={disabled}
                />
              </InnerBlock>
            )}

            {(professionalInformation.activity === "others" ||
              professionalInformation.activity === "selfEmployed" ||
              professionalInformation.activity === "professional" ||
              professionalInformation.activity === "socioOwner") && (
              <InnerBlock IEFractionParts={2}>
                <Input
                  dataTest="InputActivitySpecified"
                  onChange={e => changeInput(e, "otherActivitySpecified")}
                  value={
                    professionalInformation.otherActivitySpecified
                      ? professionalInformation.otherActivitySpecified
                      : ""
                  }
                  valid={validateInputByType["default"](
                    professionalInformation.otherActivitySpecified
                  )}
                  required
                  label={translate("KIND_OF_ACTIVITY")}
                  disabled={disabled}
                />
              </InnerBlock>
            )}

            {(professionalInformation.activity === "privateSectorEmployee" ||
              professionalInformation.activity === "publicSectorEmployee") && (
              <Fragment>
                <InnerBlock IEFractionParts={2}>
                  <Input
                    dataTest="InputOccupation"
                    onChange={e => changeInput(e, "occupation")}
                    value={professionalInformation.occupation}
                    label={translate("OCCUPATION")}
                    required
                    valid={validateInputByType["default"](
                      professionalInformation.occupation
                    )}
                    disabled={disabled}
                  />
                </InnerBlock>
                <InnerBlock span={[1, 2]} IEFractionParts={2}>
                  <Input
                    dataTest="InputCompanyName"
                    onChange={e => changeInput(e, "company", "name")}
                    value={professionalInformation.company.name}
                    required
                    label={translate("COMPANY")}
                    valid={validateInputByType["default"](
                      professionalInformation.company.name
                    )}
                    disabled={disabled}
                  />
                </InnerBlock>
              </Fragment>
            )}
          </InnerGridWrapper>
        </Block>
      </GridWrapper>
    </SectionedFormContent>
  );
}

export default ProfessionalInfo;
