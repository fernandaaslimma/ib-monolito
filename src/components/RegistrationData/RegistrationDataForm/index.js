import { connect } from "redux-zero/react";
import { combineActions } from "redux-zero/utils";

import registrationActions from "../../../actions/registration";
import notificationActions from "../../../actions/notification";
import modalActions from "../../../actions/modal";
import mfaActions from "../../../actions/mfa";
import toastrActions from "../../../actions/toastr";

import RegistrationDataForm from "./RegistrationDataForm";

export default connect(
  ({
    registrationFormData,
    dataPersonalRegistration,
    notification,
    loading,
    mfaTokenValidated,
    mfaTokenParams,
    countries,
    userInfo,
    methodChanging
  }) => ({
    registrationFormData,
    dataPersonalRegistration,
    notification,
    loading,
    mfaTokenValidated,
    mfaTokenParams,
    countries,
    userInfo,
    methodChanging
  }),
  combineActions(
    registrationActions,
    modalActions,
    notificationActions,
    mfaActions,
    toastrActions
  )
)(RegistrationDataForm);
