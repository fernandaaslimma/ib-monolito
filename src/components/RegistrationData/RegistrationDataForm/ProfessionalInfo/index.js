import ProfessionalInfo from "./ProfessionalInfo";

export default ProfessionalInfo;
