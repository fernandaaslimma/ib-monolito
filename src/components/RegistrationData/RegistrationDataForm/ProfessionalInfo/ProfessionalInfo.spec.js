import React from "react";
import { shallow } from "enzyme";
import ProfessionalInfo from "./ProfessionalInfo";
import { StateContext } from "../RegistrationDataForm";
import { configure, fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

configure({ testIdAttribute: "data-test" });

jest.mock("../../../../utils/fetchHandler");
jest.mock("../../../../utils/i18n", () => ({
  translate: id => id,
  getLanguage: () => "en-US",
  isZhCN: () => false,
  isPtBR: () => false
}));

const props = {
  onChange: jest.fn(),
  validateSection: jest.fn()
};

const e = {
  target: {
    value: "developer"
  }
};

const state = {
  formModel: {
    content: {
      professionalInformation: {
        activity: "retired",
        otherActivitySpecified: "",
        occupation: "developer",
        company: { name: "Banco BOCOM BBM", cnpj: "15114366000169" },
        admissionDate: "2015-07-19"
      }
    }
  },
  isFilled: {
    professionalInformation: true
  },
  temporaryInvalidFields: {}
};

describe("ProfessionalInfo", () => {
  it("should match snapshot with props", () => {
    expect(
      shallow(
        <StateContext.Provider value={state}>
          <ProfessionalInfo {...props} />
        </StateContext.Provider>
      )
    ).toMatchSnapshot();
  });

  it("call onChange by activity Dropdown event", () => {
    render(
      <StateContext.Provider value={state}>
        <ProfessionalInfo {...props} />
      </StateContext.Provider>
    );

    const inputs = screen.getAllByRole("textbox");
    const input = inputs[1];

    fireEvent.change(input, { target: { value: "retired" } });

    expect(props.onChange).toHaveBeenCalled();
  });
});

describe("Activity Retired", () => {
  it("should select retired as activity", () => {
    render(
      <StateContext.Provider value={state}>
        <ProfessionalInfo {...props} />
      </StateContext.Provider>
    );

    const inputs = screen.getAllByRole("textbox");
    const input = inputs[1];

    fireEvent.change(input, { target: { value: "retired" } });

    expect(props.onChange).toHaveBeenCalled();
  });

  it("should check profession input change as developer", () => {
    state.formModel = {
      content: {
        professionalInformation: {
          activity: "retired",
          otherActivitySpecified: "",
          occupation: "developer",
          company: { name: "Banco BOCOM BBM", cnpj: "15114366000169" },
          admissionDate: "2015-07-19"
        }
      }
    };
    render(
      <StateContext.Provider value={state}>
        <ProfessionalInfo {...props} />
      </StateContext.Provider>
    );

    const component = screen.getByTestId("InputOccupation");
    fireEvent.change(component, e, "occupation");
  });
});

describe("Activity Private Sector Employee", () => {
  it("call onChange by Dropdown event", () => {
    state.formModel = {
      content: {
        professionalInformation: {
          activity: "privateSectorEmployee",
          otherActivitySpecified: "",
          occupation: "developer",
          company: { name: "Banco BOCOM BBM", cnpj: "15114366000169" },
          admissionDate: "2015-07-19"
        }
      }
    };
    render(
      <StateContext.Provider value={state}>
        <ProfessionalInfo {...props} />
      </StateContext.Provider>
    );

    const inputs = screen.getAllByRole("textbox");
    const input = inputs[1];
    fireEvent.change(input, { target: { value: "privateSectorEmployee" } });

    expect(props.onChange).toHaveBeenCalled();
  });
  it("should check profession input change as developer", () => {
    state.formModel = {
      content: {
        professionalInformation: {
          activity: "privateSectorEmployee",
          otherActivitySpecified: "",
          occupation: "developer",
          company: { name: "Banco BOCOM BBM", cnpj: "15114366000169" },
          admissionDate: "2015-07-19"
        }
      }
    };
    render(
      <StateContext.Provider value={state}>
        <ProfessionalInfo {...props} />
      </StateContext.Provider>
    );

    const component = screen.getByTestId("InputOccupation");
    fireEvent.change(component, e, "occupation");

    expect(props.onChange).toHaveBeenCalled();
  });

  it("should change company input to Concrete Solutions", () => {
    state.formModel = {
      content: {
        professionalInformation: {
          activity: "privateSectorEmployee",
          otherActivitySpecified: "",
          occupation: "developer",
          company: { name: "Concrete Solutions", cnpj: "15114366000169" },
          admissionDate: "2015-07-19"
        }
      }
    };
    render(
      <StateContext.Provider value={state}>
        <ProfessionalInfo {...props} />
      </StateContext.Provider>
    );

    const component = screen.getByTestId("InputOccupation");
    fireEvent.change(component, e, "company", "name");

    expect(props.onChange).toHaveBeenCalled();
  });

  describe("Activity Public Sector Employee", () => {
    it("call onChange by Dropdown event", () => {
      render(
        <StateContext.Provider value={state}>
          <ProfessionalInfo {...props} />
        </StateContext.Provider>
      );

      const inputs = screen.getAllByRole("textbox");
      const input = inputs[1];

      fireEvent.change(input, {
        value: "publicSectorEmployee"
      });

      expect(props.onChange).toHaveBeenCalled();
    });

    it("should check profession input change as developer", () => {
      state.formModel = {
        content: {
          professionalInformation: {
            activity: "publicSectorEmployee",
            otherActivitySpecified: "",
            occupation: "developer",
            company: { name: "Banco BOCOM BBM", cnpj: "15114366000169" },
            admissionDate: "2015-07-19"
          }
        }
      };
      render(
        <StateContext.Provider value={state}>
          <ProfessionalInfo {...props} />
        </StateContext.Provider>
      );

      const component = screen.getByTestId("InputOccupation");
      fireEvent.change(component, e, "occupation");
    });

    it("should change company input to Receita Federal", () => {
      state.formModel = {
        content: {
          professionalInformation: {
            activity: "publicSectorEmployee",
            otherActivitySpecified: "",
            occupation: "developer",
            company: { name: "Receita Federal", cnpj: "15114366000169" },
            admissionDate: "2015-07-19"
          }
        }
      };
      render(
        <StateContext.Provider value={state}>
          <ProfessionalInfo {...props} />
        </StateContext.Provider>
      );

      const component = screen.getByTestId("InputOccupation");
      fireEvent.change(component, e, "occupation");

      expect(props.onChange).toHaveBeenCalled();
    });
  });

  describe("Activity Owner", () => {
    it("call onChange by Dropdown event", () => {
      render(
        <StateContext.Provider value={state}>
          <ProfessionalInfo {...props} />
        </StateContext.Provider>
      );

      const inputs = screen.getAllByRole("textbox");
      const input = inputs[1];

      fireEvent.change(input, {
        value: "owner"
      });

      expect(props.onChange).toHaveBeenCalled();
    });

    it("should change company input to Concrete Solutions", () => {
      state.formModel = {
        content: {
          professionalInformation: {
            activity: "owner",
            otherActivitySpecified: "",
            occupation: "developer",
            company: { name: "Concrete Solutions", cnpj: "15114366000169" },
            admissionDate: "2015-07-19"
          }
        }
      };
      render(
        <StateContext.Provider value={state}>
          <ProfessionalInfo {...props} />
        </StateContext.Provider>
      );

      const component = screen.getByTestId("InputCompanyName");
      fireEvent.change(component, e, "company", "name");

      expect(props.onChange).toHaveBeenCalled();
    });
  });

  describe("Activity Self-Employed", () => {
    it("call onChange by Dropdown event", () => {
      state.formModel = {
        content: {
          professionalInformation: {
            activity: "selfEmployed",
            otherActivitySpecified: null,
            occupation: "developer",
            company: { name: "Concrete Solutions", cnpj: "15114366000169" },
            admissionDate: "2015-07-19"
          }
        }
      };
      render(
        <StateContext.Provider value={state}>
          <ProfessionalInfo {...props} />
        </StateContext.Provider>
      );

      const component = screen.getByTestId("InputActivitySpecified");
      fireEvent.change(component, { value: "selfEmployed" });

      expect(props.onChange).toHaveBeenCalled();
    });

    it("should change others input to musician", () => {
      state.formModel = {
        content: {
          professionalInformation: {
            activity: "selfEmployed",
            otherActivitySpecified: "musician",
            occupation: "developer",
            company: { name: "Concrete Solutions", cnpj: "15114366000169" },
            admissionDate: "2015-07-19"
          }
        }
      };
      render(
        <StateContext.Provider value={state}>
          <ProfessionalInfo {...props} />
        </StateContext.Provider>
      );

      const component = screen.getByTestId("InputActivitySpecified");
      fireEvent.change(component, e, "otherActivitySpecified");

      expect(props.onChange).toHaveBeenCalled();
    });
  });

  describe("Activity Others", () => {
    it("call onChange by Dropdown event", () => {
      state.formModel = {
        content: {
          professionalInformation: {
            activity: "selfEmployed",
            otherActivitySpecified: "",
            occupation: "developer",
            company: { name: "Concrete Solutions", cnpj: "15114366000169" },
            admissionDate: "2015-07-19"
          }
        }
      };
      render(
        <StateContext.Provider value={state}>
          <ProfessionalInfo {...props} />
        </StateContext.Provider>
      );

      const inputs = screen.getAllByRole("textbox");
      const input = inputs[1];

      fireEvent.change(input, {
        value: "others"
      });

      expect(props.onChange).toHaveBeenCalled();
    });

    it("should change others input to musician", () => {
      state.formModel = {
        content: {
          professionalInformation: {
            activity: "others",
            otherActivitySpecified: "musician",
            occupation: "developer",
            company: { name: "Concrete Solutions", cnpj: "15114366000169" },
            admissionDate: "2015-07-19"
          }
        }
      };
      render(
        <StateContext.Provider value={state}>
          <ProfessionalInfo {...props} />
        </StateContext.Provider>
      );

      const component = screen.getByTestId("InputActivitySpecified");
      fireEvent.change(component, e, "otherActivitySpecified");

      expect(props.onChange).toHaveBeenCalled();
    });
  });

  describe("Activity Professional", () => {
    it("call onChange by Dropdown event", () => {
      state.formModel = {
        content: {
          professionalInformation: {
            activity: "professional",
            otherActivitySpecified: "",
            occupation: "developer",
            company: { name: "Concrete Solutions", cnpj: "15114366000169" },
            admissionDate: "2015-07-19"
          }
        }
      };
      render(
        <StateContext.Provider value={state}>
          <ProfessionalInfo {...props} />
        </StateContext.Provider>
      );

      const inputs = screen.getAllByRole("textbox");
      const input = inputs[1];

      fireEvent.change(input, {
        value: "others"
      });

      expect(props.onChange).toHaveBeenCalled();
    });

    it("should change others input to musician", () => {
      state.formModel = {
        content: {
          professionalInformation: {
            activity: "professional",
            otherActivitySpecified: "musician",
            occupation: "developer",
            company: { name: "Concrete Solutions", cnpj: "15114366000169" },
            admissionDate: "2015-07-19"
          }
        }
      };
      render(
        <StateContext.Provider value={state}>
          <ProfessionalInfo {...props} />
        </StateContext.Provider>
      );

      const component = screen.getByTestId("InputActivitySpecified");
      fireEvent.change(component, e, "otherActivitySpecified");

      expect(props.onChange).toHaveBeenCalled();
    });
  });

  describe("Activity Owner Partner", () => {
    it("call onChange by Dropdown event", () => {
      state.formModel = {
        content: {
          professionalInformation: {
            activity: "socioOwner",
            otherActivitySpecified: "",
            occupation: "developer",
            company: { name: "Concrete Solutions", cnpj: "15114366000169" },
            admissionDate: "2015-07-19"
          }
        }
      };
      render(
        <StateContext.Provider value={state}>
          <ProfessionalInfo {...props} />
        </StateContext.Provider>
      );

      const inputs = screen.getAllByRole("textbox");
      const input = inputs[1];

      fireEvent.change(input, {
        value: "professional"
      });

      expect(props.onChange).toHaveBeenCalled();
    });

    it("should change others input to musician", () => {
      state.formModel = {
        content: {
          professionalInformation: {
            activity: "others",
            otherActivitySpecified: "musician",
            occupation: "developer",
            company: { name: "Concrete Solutions", cnpj: "15114366000169" },
            admissionDate: "2015-07-19"
          }
        }
      };
      render(
        <StateContext.Provider value={state}>
          <ProfessionalInfo {...props} />
        </StateContext.Provider>
      );

      const component = screen.getByTestId("InputActivitySpecified");
      fireEvent.change(component, e, "otherActivitySpecified");

      expect(props.onChange).toHaveBeenCalled();
    });
  });

  describe("Activity From Home", () => {
    it("call onChange by Dropdown event", () => {
      state.formModel = {
        content: {
          professionalInformation: {
            activity: "fromHome",
            otherActivitySpecified: "",
            occupation: "developer",
            company: { name: "Receita Federal", cnpj: "15114366000169" },
            admissionDate: "2015-07-19"
          }
        }
      };
      render(
        <StateContext.Provider value={state}>
          <ProfessionalInfo {...props} />
        </StateContext.Provider>
      );

      const component = screen.getByTestId("activitySelection");
      fireEvent.change(component, { value: "fromHome" });

      expect(props.onChange).toHaveBeenCalled();
    });

    it("call onChange by Dropdown event", () => {
      state.formModel = {
        content: {
          professionalInformation: {
            activity: "",
            otherActivitySpecified: "",
            occupation: "developer",
            company: { name: "Receita Federal", cnpj: "15114366000169" },
            admissionDate: "2015-07-19"
          }
        }
      };

      expect(
        shallow(
          <StateContext.Provider value={state}>
            <ProfessionalInfo {...props} />
          </StateContext.Provider>
        )
      ).toMatchSnapshot();
    });
  });

  describe("Validation tests", () => {
    it("should validate section when toggle section container show/hide", () => {
      render(
        <StateContext.Provider value={state}>
          <ProfessionalInfo {...props} />
        </StateContext.Provider>
      );

      const component = screen.getByTestId("SectionedProfessionalInfo");
      fireEvent.click(component);

      expect(props.validateSection).toHaveBeenCalled();
    });

    it("should validate section when toggle section container show/hide", () => {
      state.temporaryInvalidFields = {
        professionalInformation: {
          company: ["company_name"],
          occupation: ["occupation"],
          otherActivitySpecified: ["kind_of_activity"]
        }
      };
      const wrapper = shallow(
        <StateContext.Provider value={state}>
          <ProfessionalInfo {...props} />
        </StateContext.Provider>
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
