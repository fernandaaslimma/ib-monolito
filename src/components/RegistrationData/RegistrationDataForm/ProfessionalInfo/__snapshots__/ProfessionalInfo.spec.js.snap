// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity Private Sector Employee Activity From Home call onChange by Dropdown event 1`] = `
<ProfessionalInfo
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "activity": "retired",
            "admissionDate": "",
            "company": Object {
              "cnpj": "",
              "name": "",
            },
            "occupation": "retired",
            "otherActivitySpecified": "",
          },
          "professionalInformation",
        ],
        Array [
          Object {
            "activity": "privateSectorEmployee",
            "admissionDate": "",
            "company": Object {
              "cnpj": "15114366000169",
              "name": "Banco BOCOM BBM",
            },
            "occupation": "privateSectorEmployee",
            "otherActivitySpecified": "",
          },
          "professionalInformation",
        ],
        Array [
          Object {
            "activity": "owner",
            "admissionDate": "",
            "company": Object {
              "cnpj": "15114366000169",
              "name": "developer",
            },
            "occupation": "",
            "otherActivitySpecified": "",
          },
          "professionalInformation",
        ],
        Array [
          Object {
            "activity": "selfEmployed",
            "admissionDate": "",
            "company": Object {
              "cnpj": "",
              "name": "",
            },
            "occupation": "",
            "otherActivitySpecified": "developer",
          },
          "professionalInformation",
        ],
        Array [
          Object {
            "activity": "others",
            "admissionDate": "",
            "company": Object {
              "cnpj": "",
              "name": "",
            },
            "occupation": "",
            "otherActivitySpecified": "developer",
          },
          "professionalInformation",
        ],
        Array [
          Object {
            "activity": "professional",
            "admissionDate": "",
            "company": Object {
              "cnpj": "",
              "name": "",
            },
            "occupation": "",
            "otherActivitySpecified": "developer",
          },
          "professionalInformation",
        ],
        Array [
          Object {
            "activity": "others",
            "admissionDate": "",
            "company": Object {
              "cnpj": "",
              "name": "",
            },
            "occupation": "",
            "otherActivitySpecified": "developer",
          },
          "professionalInformation",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  validateSection={[MockFunction]}
/>
`;

exports[`Activity Private Sector Employee Validation tests should validate section when toggle section container show/hide 1`] = `
<ProfessionalInfo
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "activity": "retired",
            "admissionDate": "",
            "company": Object {
              "cnpj": "",
              "name": "",
            },
            "occupation": "retired",
            "otherActivitySpecified": "",
          },
          "professionalInformation",
        ],
        Array [
          Object {
            "activity": "privateSectorEmployee",
            "admissionDate": "",
            "company": Object {
              "cnpj": "15114366000169",
              "name": "Banco BOCOM BBM",
            },
            "occupation": "privateSectorEmployee",
            "otherActivitySpecified": "",
          },
          "professionalInformation",
        ],
        Array [
          Object {
            "activity": "owner",
            "admissionDate": "",
            "company": Object {
              "cnpj": "15114366000169",
              "name": "developer",
            },
            "occupation": "",
            "otherActivitySpecified": "",
          },
          "professionalInformation",
        ],
        Array [
          Object {
            "activity": "selfEmployed",
            "admissionDate": "",
            "company": Object {
              "cnpj": "",
              "name": "",
            },
            "occupation": "",
            "otherActivitySpecified": "developer",
          },
          "professionalInformation",
        ],
        Array [
          Object {
            "activity": "others",
            "admissionDate": "",
            "company": Object {
              "cnpj": "",
              "name": "",
            },
            "occupation": "",
            "otherActivitySpecified": "developer",
          },
          "professionalInformation",
        ],
        Array [
          Object {
            "activity": "professional",
            "admissionDate": "",
            "company": Object {
              "cnpj": "",
              "name": "",
            },
            "occupation": "",
            "otherActivitySpecified": "developer",
          },
          "professionalInformation",
        ],
        Array [
          Object {
            "activity": "others",
            "admissionDate": "",
            "company": Object {
              "cnpj": "",
              "name": "",
            },
            "occupation": "",
            "otherActivitySpecified": "developer",
          },
          "professionalInformation",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  validateSection={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "activity": "",
            "admissionDate": "2015-07-19",
            "company": Object {
              "cnpj": "15114366000169",
              "name": "Receita Federal",
            },
            "occupation": "developer",
            "otherActivitySpecified": "",
          },
          "professionalInformation",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
/>
`;

exports[`ProfessionalInfo should match snapshot with props 1`] = `
<ProfessionalInfo
  onChange={[MockFunction]}
  validateSection={[MockFunction]}
/>
`;
